builtin/logical/ssh/path_sign.go

```diff
@@ -172,12 +213,28 @@ func (b *backend) pathSignCertificate(ctx context.Context, req *logical.Request,
 		return nil, errwrap.Wrapf("failed to parse stored CA private key: {{err}}", err)
 	}
 
+	var certValidAfter, certValidBefore uint64
+
+	// branch the logic for calculating the certificate ValidAfter & ValidBefore values
+	if ttlWasSpecified {
+		// calculate certificate ValidAfter/ValidBefore using TTL
+		now := time.Now()
+		certValidAfter = uint64(now.Add(-30 * time.Second).In(time.UTC).Unix())
+		certValidBefore = uint64(now.Add(ttl).In(time.UTC).Unix())
```

Comment:
This is the original logic to derive `ValidAfter` and `ValidBefore` from the `ttl`, it was pulled up into the sign function since there are now two ways these values can be set